lib/rubocop/cop/lint/void.rb

```diff
@@ -217,6 +217,19 @@ def autocorrect_nonmutating_send(corrector, node, suggestion)
                       end
           corrector.replace(send_node.loc.selector, suggestion)
         end
+
+        def entirely_literal?(node)
+          case node.type
+          when :array
+            node.each_value.all? { |value| entirely_literal?(value) }
+          when :hash
+            return false unless node.each_key.all?(&:literal?)
```

Comment:
technically, keys can be complex literals themselves, but I don't know if it's worth handling since it's not super common 😅  

```ruby
{
  [1, 2] => :foo
}
```

Reply:
You are right! I've updated this PR. Thanks!